### PR TITLE
[Bug-10692] fix Parameter transfer problem when recover failed task and output of subprogress

### LINKED ISSUE #10692: [Bug] [dolphinscheduler-master] Parameter transfer  problem when recover failed task and output of  subprogress
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

[Bug] [dolphinscheduler-master] Parameter transfer  problem when recover failed task and output of  subprogress

### What you expected to happen

[Bug] [dolphinscheduler-master] Parameter transfer  problem when recover failed task and output of  subprogress

### How to reproduce

[Bug] [dolphinscheduler-master] Parameter transfer  problem when recover failed task and output of  subprogress

### Anything else

[Bug] [dolphinscheduler-master] Parameter transfer  problem when recover failed task and output of  subprogress

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zwZjut**
close https://github.com/apache/dolphinscheduler/issues/10692

### CODE REVIEW

**Comment by caishunfeng** [comment]
Hi @zwZjut please resolve the conflicts, thanks.

**Comment by caishunfeng** [comment]
PTAL @ruanwenjun @lenboo

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/SubTaskProcessor.java [hunk 1]
```diff
@@ -126,11 +136,85 @@ private void updateTaskState() {
...
+                List<Property> properties = JSONUtils.toList(varPool, Property.class);
+                properties.forEach(o -> {
+                    allProperties.put(o.getProp(), o);
+                });
+            }
+        }
+        return allProperties;
+    }
+
+    private void dealFinish1() {
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/SubTaskProcessor.java:166]: Please use the original method name and remove the old method if not use.

Comment (caishunfeng) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/SubTaskProcessor.java:166]: Please add some comments and UT for this method.

**Comment by EricGao888** [comment]
@zwZjut May I ask whether you are still interested in following up with this PR?

**Pull request closed without merging**